src/Microsoft.OData.Edm/Schema/EdmOperation.cs

```diff
@@ -29,15 +29,15 @@ protected EdmOperation(string namespaceName, string name, IEdmTypeReference retu
         {
             EdmUtil.CheckArgumentNull(namespaceName, "namespaceName");
 
-            this.ReturnType = returnType;
+            this.ReturnType = returnType == null ? null : new EdmOperationReturnType(this, returnType);

```

Comment:
We should probably check to see if the passed in returnType was already an IEdmOperationReturnType